### PR TITLE
Linkedin v2 implementation

### PR DESCRIPTION
**Pull request by aleximbir**
Linkedin v2 implementation
A few more instructions for twitter, facebook, google and wordpress

### CODE REVIEW

**Comment by sviluppomania** [comment]
@aleximbir You need to change to line 62 ($ WORDPRESS_SOCIAL_LOGIN_VERSION = "3.0.2";) in the wp-social-login.php file

N.B. I get this error:

We're unable to complete your request!

Provider returned an error: unauthorized_scope_error Scope & quot; r_liteprofile & quot; is not authorized for your application

Why?

**Comment by aleximbir** [comment]
@sviluppomania 

Because the old applications created with the Linkedin v1 API no longer work with Linkedin v2 and the new applications that are created do not work with Linkedin v1, I created a new provider called 'Linkedin v2' and users can use the old Linkedin provider for old applications and 'Linkedin v2' provider for new applications, until Linkedin eliminates the possibility to use the old apps and then we can keep a single Linkedin provider.

So,
if you have an old application, you need to use 'Linkedin' provider;
if you have a new application, to need to use 'Linkedin v2' provider.

You receive this error because you are using an old application with Linkedin v2. Create a new application, enter your credentials, and try signing in with Linkedin v2 again.

File: hybridauth/callbacks/linkedin2.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+/*!
+* WordPress Social Login
+*
+* https://miled.github.io/wordpress-social-login/ | https://github.com/miled/wordpress-social-login
+*   (c) 2011-2019 Mohamed Mrassi and contributors | https://wordpress.org/plugins/wordpress-social-login/
+*/
+
+$provider_id = "LinkedIn2";
+
+require "../callback.php";
```
Comment (sviluppomania) [inline hybridauth/callbacks/linkedin2.php]: You must add **?>** To the end of the file

File: hybridauth/library/src/Provider/LinkedIn2.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        $headers = [
+            'Content-Type' => 'application/json',
+            'x-li-format'  => 'json',
+        ];
+
+        $response = $this->apiRequest('me?projection/shares?format=json', 'POST', $status, $headers);
+
+        return $response;
+    }
+}
```
Comment (sviluppomania) [inline hybridauth/library/src/Provider/LinkedIn2.php]: You must add ?> To the end of the file

File: hybridauth/callbacks/linkedin2.php (hunk 1 above)
Comment (aleximbir) [reply hybridauth/callbacks/linkedin2.php]: None of the files in the callbacks folder ends in ?> so I kept the project folder format

File: hybridauth/library/src/Provider/LinkedIn2.php (hunk 2 above)
Comment (aleximbir) [reply hybridauth/library/src/Provider/LinkedIn2.php]: None of the files in the Provider folder ends in ?> so I kept the project folder format

**Comment by mattrabe** [comment]
@aleximbir I disagree with this statement: "Because the old applications created with the Linkedin v1 API no longer work with Linkedin v2". This is not the case, as ALL applications must now use v2. v1 is permanently offline: https://engineering.linkedin.com/blog/2018/12/developer-program-updates

**Pull request closed without merging**